### PR TITLE
♻️  Use GainNode API to update video volume.

### PR DESCRIPTION
**Pull request by apatronl**
The original implementation did not work on iOS, as iOS devices leave volume under the user's physical control (see [1]), and the volume property is not settable in JavaScript. We can use the GainNode API as a workaround to this limitation.

[1] https://developer.apple.com/library/archive/documentation/AudioVideo/Conceptual/Using_HTML5_Audio_Video/Device-SpecificConsiderations/Device-SpecificConsiderations.html#//apple_ref/doc/uid/TP40009523-CH5-SW10

### CODE REVIEW

File: extensions/amp-video/validator-amp-video.protoascii [hunk 1]
```diff
@@ -185,9 +185,8 @@ tags: {  # <amp-video> in amp-story
   }
   attrs: {
     name: "volume"
-    # Disallow 0 to prevent shipping unnecessary audio track when the video is muted.
-    # Volume must be in the range [0.1, 1].
-    value_regex: "^((0?\\.[1-9]+)?|1(\\.0*)?)$"
+    # Volume between 0 and 1 (inclusive).
+    value_regex: "0(\\.[0-9]+)?|1(\\.0+)?"
```
Comment (gmajoulet) [inline extensions/amp-video/validator-amp-video.protoascii:189]: This is being done in https://github.com/ampproject/amphtml/pull/37376, I think you should be able to remove it from your PR :))

**Review by gmajoulet — APPROVED** [review]

File: extensions/amp-video/validator-amp-video.protoascii (hunk 1 above)
Comment (apatronl) [reply extensions/amp-video/validator-amp-video.protoascii]: oh cool, thanks!

**Comment by gmajoulet** [comment]
Can you fix the [lint checks](https://circleci.com/gh/ampproject/amphtml/430001?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) before we can merge? Thank you!

**Review by gmajoulet — APPROVED** [review]

**Comment by gmajoulet** [comment]
```
checks.js Running amp lint --local_changes...
[18:48:05] Using task file project/amp.js
[18:48:05] Starting 'lint'...
[18:48:08] 
extensions/amp-story/1.0/media-pool.js
  903:60  error  'webkitAudioContext' is not defined  no-undef
```

I think you might need to add a `/*global*/ in front of `webkitAudioContext`

**Pull request merged**